parsing/location.ml

```diff
@@ -258,29 +263,46 @@ let print_loc ppf loc =
       fprintf ppf "Characters %i-%i"
               loc.loc_start.pos_cnum loc.loc_end.pos_cnum
   end else begin
-    fprintf ppf "%s%a%s%i" msg_file print_filename file msg_line line;
+    fprintf ppf "%s@{<loc>%a%s%i@}" msg_file print_filename file msg_line line;
     if startchar >= 0 then
       fprintf ppf "%s%i%s%i" msg_chars startchar msg_to endchar
   end
 ;;
 
 let print ppf loc =
+  setup_colors ();
   if loc.loc_start.pos_fname = "//toplevel//"
   && highlight_locations ppf [loc] then ()
-  else fprintf ppf "%a%s@." print_loc loc msg_colon
+  else fprintf ppf "@{<loc>%a@}%s@." print_loc loc msg_colon
+;;
+
+let error_prefix = "Error:"
+let warning_prefix = "Warning"
+
+let print_error_prefix ppf () =
+  setup_colors ();
+  pp_open_tag ppf "error";
```

Comment:
Is there a particular reason for not using a format here? (I don't mind either the format or plain-`pp` styles, but mixing both is a bit surprising)